sim/src/mechanics/driving.rs

```diff
@@ -97,19 +97,62 @@ impl DrivingSimState {
         ctx: &mut Ctx,
     ) -> Option<CreateCar> {
         let first_lane = params.router.head().as_lane();
-        let start_dist = params.router.get_path().get_req().start.dist_along();
+        let mut start_dist = params.router.get_path().get_req().start.dist_along();
+        if let Some(ref p) = params.maybe_parked_car {
+            // If we're exiting a driveway, make the front of the vehicle wind up in the correct
+            // spot after the driveway. We could attempt to do this when we create the PathRequest,
+            // but it's complicated to adjust the position correctly, and this is the only place
+            // that needs to know.
+            if !matches!(p.spot, ParkingSpot::Onstreet(_, _)) {
+                start_dist += params.vehicle.length;
+                // TODO Should we also adjust the request?
+                if start_dist > ctx.map.get_l(first_lane).length() {
+                    // I'm not sure if this can happen in practice; we'll find out if so.
+                    panic!(
+                        "At {}, {} needs to exit a driveway at {} on {}, but they overflow",
+                        now, params.vehicle.id, start_dist, first_lane
+                    );
+                }
+            }
+        }
 
-        // First handle any of the intermediate queues, failing fast.
-        let mut blocked_start_indices = Vec::new();
-        for pos in params.router.get_path().get_blocked_starts() {
-            if let Some(idx) = self.queues[&Traversable::Lane(pos.lane())].can_block_from_driveway(
-                pos,
+        // First handle any of the intermediate queues, failing fast. Record the position of the
+        // blockage's front and the index in that queue.
+        let mut blocked_starts: Vec<(Position, usize)> = Vec::new();
+        for lane in params.router.get_path().get_blocked_starts() {
+            // This buffer makes sure other vehicles can enter the queue behind a blockage very
+            // close to the start of the lane and not spillover.
+            let pos = match params
+                .router
+                .get_path()
+                .get_req()
+                .start
+                .equiv_pos(lane, ctx.map)
+                .buffer_dist(MAX_CAR_LENGTH + FOLLOWING_DISTANCE, ctx.map)
+            {
+                Some(pos) => pos,
+                None => {
+                    // TODO Loss of some simulation realism. We could also ban this upfront in
```

Comment:
There's a case near b3810 in greenlake that I can repro easily, where the lanes on either side of a service road have about a 10 meter difference, causing a problem. Root problem here is actually `equiv_pos` not being smart. I'll look into making it symmetric by projecting a little perpendicular line and finding the intersection.